### ISSUE TITLE
Using double quotes `"` in `$((...))` is a bash-ism

### ISSUE THREAD

**Issue by emanuele6** [koalaman/shellcheck#2055]
#### For bugs
- Rule Id (if any, e.g. SC1000): 
- My shellcheck version (`shellcheck --version` or "online"): online
- [] The rule's wiki page does not already cover this (e.g. https://shellcheck.net/wiki/SC2086)
- [] I tried on https://www.shellcheck.net/ and verified that this is still a problem on the latest commit

#### For new checks and feature suggestions
- [x] https://www.shellcheck.net/ (i.e. the latest commit) currently gives no useful warnings about this
- [x] I searched through https://github.com/koalaman/shellcheck/issues and didn't find anything related

#### Here's a snippet or screenshot that shows the problem:

```sh
#!/bin/sh

echo "$(("$(date +%u)" * 2))"
```
#### Here's what shellcheck currently says:

Nothing

#### Here's what I wanted or expected to see:

`bash` will substitute `"$(date +%u)"` with `3` → `"$((3 * 2))"` that is valid.
```
6
```
`dash` will only substitute `$(date +%u)` with `3` → `"$(("3" * 2))" which isn't valid:
```
file.sh: 3: arithmetic expression: expecting primary: "2 * "3""
```

`shellcheck` should, at least, warn me about using `"` in a `$((...))` if the shell is not `bash`.
It could also suggest me to remove them this `"`s.
